tools/eosiocpp.in

```diff
@@ -51,7 +51,12 @@ function build_contract {
 
     done
 
-    ($PRINT_CMDS; @WASM_LLVM_LINK@ -only-needed -o $workdir/linked.bc $workdir/built/* \
+    linkfiles=''
+    for file in `ls -t $workdir/built/ | sort -r`; do
```

Comment:
This isn't fixing the underlying issue. Your original fix, where you construct `$linkfiles` ordered by how they are given on command line, is preferable I believe.